pkg/plugin/v1/api.go

```diff
@@ -0,0 +1,149 @@
+
+	apiScaffolder scaffold.API
+}
+
+var _ plugin.CreateAPI = &createAPIPlugin{}
+
+func (p createAPIPlugin) UpdateContext(ctx *plugin.PluginContext) {
+	ctx.Description = `Scaffold a Kubernetes API by creating a Resource definition and / or a Controller.
+
+create resource will prompt the user for if it should scaffold the Resource and / or Controller.  To only
+scaffold a Controller for an existing Resource, select "n" for Resource.  To only define
+the schema for a Resource without writing a Controller, select "n" for Controller.
+
+After the scaffold is written, api will run make on the project.
+`
+	ctx.Examples = fmt.Sprintf(`  # Create a frigates API with Group: ship, Version: v1beta1 and Kind: Frigate
+  %s create api --group ship --version v1beta1 --kind Frigate
+
+  # Edit the API Scheme
+  nano api/v1beta1/frigate_types.go
+
+  # Edit the Controller
+  nano controllers/frigate/frigate_controller.go
+
+  # Edit the Controller Test
+  nano controllers/frigate/frigate_controller_test.go
+
+  # Install CRDs into the Kubernetes cluster using kubectl apply
+  make install
+
+  # Regenerate code and run against the Kubernetes cluster configured by ~/.kube/config
+  make run
+`, p.commandName)
+}
+
+func (p *createAPIPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&p.runMake, "make", true,
+		"if true, run make after generating files")
+	fs.BoolVar(&p.apiScaffolder.DoResource, "resource", true,
+		"if set, generate the resource without prompting the user")
+	p.resourceFlag = fs.Lookup("resource")
+	fs.BoolVar(&p.apiScaffolder.DoController, "controller", true,
+		"if set, generate the controller without prompting the user")
+	p.controllerFlag = fs.Lookup("controller")
+	if os.Getenv("KUBEBUILDER_ENABLE_PLUGINS") != "" {
+		fs.StringVar(&p.pattern, "pattern", "",
+			"generates an API following an extension pattern (addon)")
+	}
+	p.apiScaffolder.Resource = pluginutil.ResourceForFlags(fs)
+}
+
+func (p *createAPIPlugin) Run() error {
+	p.runAddAPI()
+	return nil
+}
+
+// APICmd represents the resource command
+func (p *createAPIPlugin) runAddAPI() {
+	pluginutil.DieIfNoProject()
+
+	switch strings.ToLower(p.pattern) {
+	case "":
+		// Default pattern
+
+	case "addon":
+		p.apiScaffolder.Plugins = append(p.apiScaffolder.Plugins, &addon.Plugin{})
+
+	default:
+		log.Fatalf("unknown pattern %q", p.pattern)
+	}
+
+	if err := p.apiScaffolder.Validate(); err != nil {
+		log.Fatalln(err)
+	}
+
+	reader := bufio.NewReader(os.Stdin)
+	if !p.resourceFlag.Changed {
+		fmt.Println("Create Resource [y/n]")
+		p.apiScaffolder.DoResource = util.Yesno(reader)
+	}
+
+	if !p.controllerFlag.Changed {
+		fmt.Println("Create Controller [y/n]")
+		p.apiScaffolder.DoController = util.Yesno(reader)
+	}
+
+	fmt.Println("Writing scaffold for you to edit...")
+
+	if err := p.apiScaffolder.Scaffold(); err != nil {
+		log.Fatal(err)
+	}
+
+	if err := p.postScaffold(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (p *createAPIPlugin) postScaffold() error {
```

Comment:
You changed this method name for the init plugins, were you trying to avoid the "scaffold" word? If so you still have one here. `postAPI`?

Reply:
Its an unnecessary change so I'm switching back to `postScaffold`.